src/ensembl/src/content/app/species/components/species-main-view/SpeciesMainViewStats.tsx

```diff
@@ -0,0 +1,217 @@
+                box: styles.pointerBox,
+                pointer: styles.pointerBoxPointer
+              }}
+            >
+              <ViewInApp links={exampleLinks} />
+            </PointerBox>
+          )}
+        </span>
+      </span>
+    </div>
+  );
+};
+
+const getCollapsedContent = (statsSection: StatsSection) => {
+  const { primaryStats, secondaryStats, section, exampleLinks } = statsSection;
+  const { title, exampleLinkText } = sectionGroupsMap[section];
+
+  return (
+    <div className={styles.collapsedContent}>
+      <span className={styles.title}>{title}</span>
+      {primaryStats && (
+        <div className={styles.primary}>
+          <span className={styles.primaryValue}>
+            {primaryStats.primaryValue}
+          </span>
+          <span className={styles.primaryUnit}>{primaryStats.primaryUnit}</span>
+        </div>
+      )}
+
+      {secondaryStats && (
+        <div className={styles.secondary}>
+          <span className={styles.secondaryValue}>
+            {secondaryStats.secondaryValue}
+          </span>
+          <span className={styles.secondaryUnit}>
+            {secondaryStats.secondaryUnit}
+          </span>
+        </div>
+      )}
+
+      <ExampleLinks
+        exampleLinks={exampleLinks}
+        exampleLinkText={exampleLinkText}
+      />
+    </div>
+  );
+};
+
+const getExpandedContent = (statsSection: StatsSection) => {
+  const { groups, exampleLinks, section } = statsSection;
+  const { exampleLinkText } = sectionGroupsMap[section];
+  return (
+    <div className={styles.expandedContent}>
+      {groups.map((group, group_index: number) => {
+        const { title, stats } = group;
+        const primaryStatsClassName = classNames(styles.primaryStats, {
+          [styles.primaryStatsNoLink]: exampleLinks
+            ? !Object.keys(exampleLinks).length
+            : true
+        });
+        const primaryOrSecondaryClassName =
+          group_index === 0 ? primaryStatsClassName : styles.secondaryStats;
+        return (
+          <div key={group_index} className={primaryOrSecondaryClassName}>
+            <span className={styles.title}>{title}</span>
+            <div className={styles.stats}>
+              {stats.map((groupStats, group_index: number) => {
+                return (
+                  <div key={group_index}>
+                    {groupStats.map((stat, stat_index: number) => {
+                      return <SpeciesStats key={stat_index} {...stat} />;
+                    })}
+                  </div>
+                );
+              })}
+            </div>
+            {group_index === 0 && (
+              <ExampleLinks
+                exampleLinks={exampleLinks}
+                exampleLinkText={exampleLinkText}
+              />
+            )}
+          </div>
+        );
+      })}
+    </div>
+  );
+};
+
+const SpeciesMainViewStats = (props: Props) => {
+  useEffect(() => {
+    if (!props.genomeStats && props.exampleFocusObjects?.length) {
```

Comment:
Ouch! This is a very confusing dependency. Fetching genome stats should not depend on whether or not we have fetched example focus objects.

Reply:
That's right. Removed `props.exampleFocusObjects?.length` from here and moved it to the if statement below.

Reply:
Example links are getting generated while the stats are getting processed. This is why I'm trying to make sure the exampleObjects are available before fetching the genomeStats.